src/test/regress/expected/coordinator_shouldhaveshards.out

```diff
@@ -188,6 +188,28 @@ NOTICE:  executing the command locally: SELECT count(*) AS count FROM coordinato
    100
 (1 row)
 
+ROLLBACK;
+BEGIN;
+SET citus.enable_repartition_joins TO ON;
+-- trigger local execution
+SELECT y FROM test WHERE x = 1;
+NOTICE:  executing the command locally: SELECT y FROM coordinator_shouldhaveshards.test_1503000 test WHERE (x OPERATOR(pg_catalog.=) 1)
+ y
+---------------------------------------------------------------------
+ 1
+(1 row)
+
+-- this will use local execution, it should work fine without giving segfault.
+SELECT count(*) FROM test t1, test t2 WHERE t1.x = t2.y;
+NOTICE:  executing the command locally: SELECT worker_hash_partition_table  (xxxxxxx, 1, 'SELECT x FROM coordinator_shouldhaveshards.test_1503000 t1 WHERE true', 'x', 'integer'::regtype, '{-2147483648,-1073741824,0,1073741824}'::integer[])
+NOTICE:  executing the command locally: SELECT worker_hash_partition_table  (xxxxxxx, 4, 'SELECT x FROM coordinator_shouldhaveshards.test_1503003 t1 WHERE true', 'x', 'integer'::regtype, '{-2147483648,-1073741824,0,1073741824}'::integer[])
+NOTICE:  executing the command locally: SELECT worker_hash_partition_table  (xxxxxxx, 1, 'SELECT y FROM coordinator_shouldhaveshards.test_1503000 t2 WHERE true', 'y', 'integer'::regtype, '{-2147483648,-1073741824,0,1073741824}'::integer[])
```

Comment:
does this follow a deterministic order?

Reply:
Yeah it seems so. The tasks are generated in a deterministic order and in local execution we execute them one by one starting from the beginning hence that is also deterministic.

Reply:
Even though it passes in my local, weirdly it fails with in circleci:

```diff
+ERROR:  COPY file signature not recognized
+CONTEXT:  while executing command on localhost:57637
```

https://circleci.com/gh/citusdata/citus/113058

Not sure how this PR can cause such a failure.

Reply:
Though we didn't have such a test before, so it might be that it is an existing issue.

Reply:
One guess is that the we are trying to read the file (via a worker_fetch_partition_file from a worker) before it is closed.

Reply:
Looking at `worker_hash_partition_table` it seems that we are closing the files with `ClosePartitionFiles` in a sync way I guess. The error is thrown at:

```
		if (CopyGetData(cstate, readSig, 11, 11) != 11 ||
			memcmp(readSig, BinarySignature, 11) != 0)
			ereport(ERROR,
					(errcode(ERRCODE_BAD_COPY_FILE_FORMAT),
					 errmsg("COPY file signature not recognized")));
```

So is your guess that we can't read read from the file therefore it reads 0 byte, which is not the expected(11). And this error is thrown.

Reply:
If this ends up taking too much time we could set localExecutionSupported = false in ExecuteTaskListOutsideTransaction

Reply:
This one is kind of hard to fix because it fails on circleci and also sometimes when I run the whole check-multi schedule, not just a few of the tests.

Probably setting localExecutionSupported = false makes sense here. Also if we wont have visibility problems, we can still use local execution even though we open a connection to localhost.